frontend/ui/widget/spinwidget.lua

```diff
@@ -52,6 +52,7 @@ local SpinWidget = InputContainer:new{
 }
 
 function SpinWidget:init()
+    self._value = self.value -- used to enable ok_button, self.value may be changed in extra callback
```

Comment:
You could name it with an explicite name `self.currently_applied_value` or `self.original_value`.
It would make the stuff below reads as natural language :)